docs/modules/ROOT/pages/api/erc20.adoc

```diff
@@ -158,35 +127,42 @@ Implementation of the <<IERC20,IERC20>> interface.
 --
 
 [.contract-index]
-.External functions
+.Embeddable functions
```

Comment:
No and yes 👍

Reply:
why `Embeddable` and not `Public` or `External`?

Reply:
The rationale is that they are not really public, unless you embed them (you can use them without embedding)

Reply:
the thing is that you can embed any function, not only those. The fact that they're not external unless you embed or manually expose them goes without saying to me -- we can assume readers understand how Cairo works.

as mentioned below, my concern is that this leaves us without good categorization, reflected in the camelCase methods not being under "Embeddable"

Reply:
the thing is that you can embed any function, not only those. The fact that they're not external unless you embed or manually expose them goes without saying to me -- we can assume readers understand how Cairo works.

as mentioned below, my concern is that this leaves us without good categorization, reflected in the camelCase methods not being under "Embeddable"

Reply:
> the thing is that you can embed any function, not only those

Not really, you can only embed impl annotated with `#[starknet::embeddable]`, and those are the ones we are presenting in the Embeddable section.

> The fact that they're not external unless you embed or manually expose them goes without saying to me -- we can assume readers understand how Cairo works. 

To me saying they are external is not accurate, because they are not, even if they are designed to potentially become external (that's why embeddable made sense to me, but maybe another word is better, naming things :( ).

You can use the component, use the implementation, and not make the function external. Using the name external could create some confusion among readers IMO.

Reply:
> #[starknet::embeddable]

The compiler is the one generating the impl from the `embeddable_as` annotation, even if we are not doing that explicitly in the codebase.

Reply:
> Not really, you can only embed impl annotated with `#[starknet::embeddable]`, and those are the ones we are presenting in the Embeddable section.

you're right

> To me saying they are external is not accurate, because they are not, even if they are designed to potentially become external

i guess my issue with embeddable is that users might simply use them as building blocks

> naming things :(

:'(